docs/developer/dependency_resolution.md

```diff
@@ -99,6 +99,10 @@ depends on a `FooService` and because of that, any provider of a `FooStore` must
 the `FooMaintenanceExtension`. The fact that `CosmosFooStoreExtension` provides a `FooStore` is declared using
 the `@Provides` annotation.
 
+To test such classes, use the appropriate JUnit extension:
```

Comment:
Perhaps moved to a separate sub-heading/section e.g. `Service Injection within tests.`